cmake/modules/dts.cmake

```diff
@@ -5,6 +5,12 @@ include_guard(GLOBAL)
 include(extensions)
 include(python)
 include(boards)
+
+# Include board specific device-tree flags before parsing.
+foreach(dir IN LISTS BOARD_DIRECTORIES)
+  include("${dir}/pre_dt_board.cmake" OPTIONAL)
```

Comment:
this suffers the same issue as described here: https://github.com/zephyrproject-rtos/zephyr/pull/87821#discussion_r2018788625